tests/acceptance/30_generic_methods/directory_absent.cf

```diff
@@ -0,0 +1,129 @@
+      create => "true";
+
+    #----
+
+    "${directory}/subdir3/."
+      create => "true";
+
+    "${directory}/subdir3/subfile3"
+      create => "true";
+
+    "${directory}/subdir3/subsubdir3/."
+      create => "true";
+
+    "${directory}/subdir3/subsubdir3/subsubfile3"
+      create => "true";
+
+}
+
+#######################################################
+
+bundle agent test
+{
+  methods:
+    "ph0" usebundle => directory_absent("${init.directory}/subdir0", "true");
+    "ph1" usebundle => directory_absent("${init.directory}/subdir1/", "true");
+    "ph2" usebundle => directory_absent("${init.directory}/subdir2", "true");
+    "ph3" usebundle => directory_absent("${init.directory}/subdir3", "false");
+    "ph4" usebundle => directory_absent("${init.directory}/subdir4", "true");
+    "ph5" usebundle => directory_absent("${init.directory}/subdir5", "false");
+    "ph6" usebundle => directory_absent("${init.directory}/file", "true");
+    "ph7" usebundle => directory_absent("${init.directory}/file", "false");
+}
+
+#######################################################
+
+bundle agent check
+{
+  classes:
+    "file_exists_0" expression => fileexists("${init.directory}/subdir0");
+    "file_exists_1" expression => fileexists("${init.directory}/subdir1");
+    "file_exists_2" expression => fileexists("${init.directory}/subdir2");
+    "file_exists_3" expression => fileexists("${init.directory}/subdir3");
+    "file_exists_4" expression => fileexists("${init.directory}/subdir4");
+    "file_exists_5" expression => fileexists("${init.directory}/subdir5");
+    "file_exists_6" expression => fileexists("${init.directory}/file");
+    "file_exists_7" expression => fileexists("${init.directory}/file");
+    "ok_0"          expression => "!file_exists_0.directory_absent_${init.directory_canon_0}_repaired.!directory_absent_${init.directory_canon_0}_error";
+    "ok_1"          expression => "!file_exists_1.directory_absent_${init.directory_canon_1}_repaired.!directory_absent_${init.directory_canon_1}_error";
+    "ok_2"          expression => "!file_exists_2.directory_absent_${init.directory_canon_2}_repaired.!directory_absent_${init.directory_canon_2}_error";
+    "ok_3"          expression => "file_exists_3.!directory_absent_${init.directory_canon_3}_repaired.directory_absent_${init.directory_canon_3}_error";
+    "ok_4"          expression => "!file_exists_4.directory_absent_${init.directory_canon_4}_kept.!directory_absent_${init.directory_canon_4}_error";
+    "ok_5"          expression => "!file_exists_5.directory_absent_${init.directory_canon_5}_kept.!directory_absent_${init.directory_canon_5}_error";
+    "ok_6"          expression => "file_exists_6.!directory_absent_${init.directory_canon_6}_kept.directory_absent_${init.directory_canon_6}_error";
+    "ok_7"          expression => "file_exists_7.!directory_absent_${init.directory_canon_7}_kept.directory_absent_${init.directory_canon_7}_error";
```

Comment:
There is a file with the same name, I don't think we can ignore it. We may see it as:

```
# rmdir LICENSE 
rmdir: failed to remove 'LICENSE': Not a directory
```